### ISSUE TITLE
Deprecate util.unique_rows in favor of numpy.unique

### ISSUE THREAD

**Issue by lagru** [scikit-image/scikit-image#4145]
I noticed that the function [skimage.util.unique_rows](https://scikit-image.org/docs/dev/api/skimage.util.html?highlight=unique_rows#skimage.util.unique_rows) can be trivially replaced with `np.unique(..., axis=0)` [[ref]](https://docs.scipy.org/doc/numpy/reference/generated/numpy.unique.html) since NumPy version 1.13 unless I'm missing something (related tests pass). Since that seems to be the minimal version required by scikit-image anyway I propose to depreciate that function.

An alternate course would be to just simplify the internals of that function.

**Comment by hmaarrfk** [comment]
The expressiveness of that function seems useful for people writing algorithms.

https://github.com/search?l=Python&q=unique_rows&type=Code

It sometimes gets confusing thinking of axes, and ordering.

I do agree that the logic is a little convoluted, and pushing that off to numpy might be a good cleanup.

**Comment by jni** [comment]
Indeed, that function predates 1.13, and is less useful now. But imho @hmaarrfk's search (which we should probably get in the habit of doing every time we discuss a deprecation, thanks @hmaarrfk for doing that!) is sufficient to say that we should keep that function around indefinitely, though perhaps we can stop using it internally, and maybe even soft-deprecate it (mention it as deprecated in the docstring, with a mention of the preferred alternative). (It'd be interesting to compare the performance of the two options, though.)

**Comment by JDWarner** [comment]
I'm interested if the `unique(..., axis=X)` motif improves performance over our implementation which predates the convenience kwarg.  If it does, we should wrap the new motif and simplify the function, perhaps add a note to the docstring about the functionality in NumPy.

**Comment by lagru** [comment]
@JDWarner Good question. `np.unique` is noticeably slower:

```python
import numpy as np
from skimage.util import unique_rows

image = np.arange(60000).reshape((200, 300))
image = np.concatenate([image, image], axis=0)

%timeit unique_rows(image)
%timeit np.unique(image, axis=0)
800 µs ± 35.2 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
2.94 ms ± 32.1 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

And there is another implication!

`plt.imshow(plt.imshow(unique_rows(image, axis=0)))` gives this:

![uniquerows](https://user-images.githubusercontent.com/20140352/64519125-d097b580-d2f3-11e9-8601-49a33df8f079.png)
while `plt.imshow(plt.imshow(np.unique(image, axis=0)))` will return this:
![npunique](https://user-images.githubusercontent.com/20140352/64519112-cd9cc500-d2f3-11e9-904a-2b8531c28816.png)
